contracts/core/src/solve/Inbox.sol

```diff
@@ -104,6 +188,23 @@ contract Inbox is ReentrancyGuardUpgradeable {
         }
     }
 
+    function _cancelRequest(Solve.Request storage req) internal {
+        // Refund deposits
+        uint256 length = req.deposits.length;
+        for (uint256 i = 0; i < length; i++) {
+            Solve.Deposit memory deposit = req.deposits[i];
+            if (deposit.isNative) {
+                (bool success,) = payable(msg.sender).call{ value: deposit.amount }("");
+                if (!success) revert TransferFailed();
+            } else {
+                IERC20(deposit.token).safeTransfer(msg.sender, deposit.amount);
+            }
+        }
+
+        // Delete request
+        delete _requests[req.id];
```

Comment:
Was thinking a little more about this. Feel like it will make solver spec much easier if it can just query source inbox for status of of a request ID. Rather than indexing events and maintaining state in db. 

We can work with corver on this when he starts. But I propose:

- Let's keep things simple for noew and just keep requests, updating their status
- If it turns out we don't need view funcs for this, we can delete
- If it turns out we do, but we want gas rebase for deletions, we can move status to another mapping